framework/include/utils/InputParameters.h

```diff
@@ -1830,3 +1851,79 @@ InputParameters::have_parameter(std::string_view name_in) const
 
   return Parameters::have_parameter<T>(name);
 }
+
+template <typename T>
+void
+InputParameters::transferParam(const InputParameters & source_params,
+                               std::string name_in,
+                               std::string new_description)
+{
+  const auto name = source_params.checkForRename(std::string(name_in));
+  if (!source_params.have_parameter<T>(name) && !source_params.hasCoupledValue(name))
+    mooseError(
+        "The '",
+        name_in,
+        "' parameter could not be transferred because it does not exist in the source parameters");
+
+  std::string description =
+      new_description.empty() ? source_params.getDescription(name) : new_description;
+
+  if (source_params.isParamRequired(name))
+  {
+    // Use the name_in that was specified in the transfer, not the potentially deprecated name
+    // Check for a variable parameter
+    if (source_params.hasCoupledValue(name))
+      addRequiredCoupledVar(name, description);
```

Comment:
```suggestion
      addRequiredCoupledVar(name_in, description);
```
? This would be consistent with your other code